04-cs-fundamentals/classroom/Sorting.md

```diff
@@ -116,35 +177,49 @@ def insertion_sort(array, length)
   end
 end
 ```
-<strong>Example:</strong> Consider the initial unsorted array [99, 45, 35, 40, 16, 50, 11, 7, 90]. Here's what it looks like before and after each iteration of the outer loop. The sorted sub-array is **bolded**</br>
-&nbsp;&nbsp;&nbsp;&nbsp;[**99**, 45, 35, 40, 16, 50, 11, 7, 90]</br>
-&nbsp;&nbsp;&nbsp;&nbsp;[**45, 99**, 35, 40, 16, 50, 11, 7, 90]</br>
-&nbsp;&nbsp;&nbsp;&nbsp;[**35, 45, 99**, 40, 16, 50, 11, 7, 90]</br>
-&nbsp;&nbsp;&nbsp;&nbsp;[**35, 40, 45, 99**, 16, 50, 11, 7, 90]</br>
-&nbsp;&nbsp;&nbsp;&nbsp;[**16, 35, 40, 45, 99**, 50, 11, 7, 90]</br>
-&nbsp;&nbsp;&nbsp;&nbsp;[**16, 35, 40, 45, 50, 99**, 11, 7, 90]</br>
-&nbsp;&nbsp;&nbsp;&nbsp;[**11, 16, 35, 40, 45, 50, 99**, 7, 90]</br>
-&nbsp;&nbsp;&nbsp;&nbsp;[**7, 11, 16, 35, 40, 45, 50, 99**, 90]</br>
-&nbsp;&nbsp;&nbsp;&nbsp;[**7, 11, 16, 35, 40, 45, 50, 90, 99**]</br>
-
-<strong>Analysis:</strong> In sorting the most expensive part is a comparison of two elements. Insertion sort algorithm requires
+
+**Example:** Consider the initial unsorted array [99, 45, 35, 40, 16, 50, 11, 7, 90]. Here's what it looks like before and after each iteration of the outer loop. The sorted sub-array is **bolded**
+
+| Iteration | Array |
+|-- |-- |
+| 1. | [**99**, 45, 35, 40, 16, 50, 11, 7, 90] |
+| 2. | [**45, 99**, 35, 40, 16, 50, 11, 7, 90] |
+| 3. | [**35, 45, 99**, 40, 16, 50, 11, 7, 90] |
+| 4. | [**35, 40, 45, 99**, 16, 50, 11, 7, 90] |
+| 5. | [**16, 35, 40, 45, 99**, 50, 11, 7, 90] |
+| 6. | [**16, 35, 40, 45, 50, 99**, 11, 7, 90] |
+| 7. | [**11, 16, 35, 40, 45, 50, 99**, 7, 90] |
+| 8. | [**7, 11, 16, 35, 40, 45, 50, 99**, 90] |
+| 9. | [**7, 11, 16, 35, 40, 45, 50, 90, 99**] |
+
+**Analysis:** In sorting the most expensive part is a comparison of two elements. Insertion sort algorithm requires
+
 - 0 comparisons to insert the first element
 - 1 comparison to insert the second element
 - 2 comparisons to insert the third element
 - ... and so on
 - *n-1* comparisons (worst case) to insert the last element
-Overall, this is *1 + 2 + 3 + ... + (n-1)* = *O(n^2)*
 
-The insertion sort has a complexity of O(n^2). Unlike selection sort, insertion sort has a best case time complexity of O(n) when the items are already sorted. In other words, insertion sort runs in linear time on a nearly sorted list of elements.
+Overall, this is *1 + 2 + 3 + ... + (n-1)* = *O(n<sup>2</sup>)*
+
+The insertion sort has a complexity of O(n<sup>2</sup>). Unlike selection sort, insertion sort has a best case time complexity of O(n) when the items are already sorted. In other words, insertion sort runs in linear time on a nearly sorted list of elements.
```

Comment:
```suggestion
The insertion sort has a complexity of O(n<sup>2</sup>). Unlike selection sort, insertion sort has a best case time complexity of O(n) when the items are already sorted. In other words, insertion sort runs in nearly linear time on a nearly sorted list of elements.
```

Reply:
Maybe also add a note about how for small arrays (I haven't actually seen a good definition of what "small" means but I imagine it depends on the architecture) Insertion Sort is often the fastest sort.  I feel like that's usually why it comes up.